rust/datafusion/src/datasource/datasource.rs

```diff
@@ -24,6 +24,7 @@ use arrow::record_batch::RecordBatch;
 
 use crate::execution::error::Result;
 
+/// Iterator of record batches, wrapped in a mutex
```

Comment:
I guess this is a good time to explain _why_ they are wrapped in mutexes. This is because the iterators are mutable and I want to share them between threads to allow for parallel query execution eventually.

Reply:
Rephrased this one.